### ISSUE TITLE
Qwen Image pipeline, Various

### ISSUE THREAD

**Issue by dxqb** [huggingface/diffusers#12295]
### Describe the bug

First of all, thank you for providing this pipeline! It makes it much easier to build tools on top of these models.
While doing that, I ran across various issues, big and small. I'll list them here - let me know if some of them should be a separate issue.

1. 
Here:
https://github.com/huggingface/diffusers/blob/fc337d585309c4b032e8d0180bea683007219df1/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L212C9-L212C74
The system prompt is removed from the hidden states.

The encoded hidden state of:
```
"<|im_start|>system\nDescribe the image by detailing the color, shape, size, texture, quantity, text, spatial relationships of the objects and background:<|im_end|>\n<|im_start|>user\n{}<|im_end|>\n<|im_start|>assistant\n"
```

is turned into the encoded hidden state of:
`{}<|im_end|>\n<|im_start|>assistant\n`

I cannot find in the Qwen technical report that the hidden_states of the system prompt were removed during training, but I assume that someone involved with Qwen has provided this information.
But then, it's surprising that the remaining part of the system prompt after the user prompt is *not* removed: `{}<|im_end|>\n<|im_start|>assistant\n` remains in the encoder hidden state passed to the transformer.
Can you confirm that this is correct, and as the model was trained?

2.
Unless I'm missing something, the splitting of the hidden states per batch sample here https://github.com/huggingface/diffusers/blob/fc337d585309c4b032e8d0180bea683007219df1/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L211 and then stacking them back into batched hidden states, doesn't do anything. The end result is the same as the input.
The only thing it seems to do, is to zero the masked tokens. This should be irrelevant - because they are masked. But even if not, the same thing can be achieved by

`prompt_embeds = hidden_states[:, drop_idx:,:] * attention_mask.unsqueeze(-1) `

What was the intention of this code?

3.
This seems unnecessary:
https://github.com/huggingface/diffusers/blob/fc337d585309c4b032e8d0180bea683007219df1/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L257
The prompt is already truncated by the tokenizer here:
https://github.com/huggingface/diffusers/blob/fc337d585309c4b032e8d0180bea683007219df1/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L203

If the intention is to have the option to truncate it further, why have the first truncation hardcoded to 1024 here, and then do it a second time?
https://github.com/huggingface/diffusers/blob/fc337d585309c4b032e8d0180bea683007219df1/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L175

4.
Documentation of these parameters is missing:
https://github.com/huggingface/diffusers/blob/fc337d585309c4b032e8d0180bea683007219df1/src/diffusers/models/transformers/transformer_qwenimage.py#L554

5. 
The type hint of `img_shapes` is a list of tuples.
But actually, a list *of a list* of tuples is passed here: https://github.com/huggingface/diffusers/blob/fc337d585309c4b032e8d0180bea683007219df1/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L612
Which is then converted from a *list of a list of tuples* into a *list of tuples* here:
https://github.com/huggingface/diffusers/blob/fc337d585309c4b032e8d0180bea683007219df1/src/diffusers/models/transformers/transformer_qwenimage.py#L206
but what's the intention here? Why is it even a list, do you plan to support different shapes in the same batch? Why is it a list of a list?

6.
These defaults are incorrect for Qwen:
https://github.com/huggingface/diffusers/blob/fc337d585309c4b032e8d0180bea683007219df1/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L620
Compare here: https://huggingface.co/Qwen/Qwen-Image/blob/main/scheduler/scheduler_config.json
I guess the defaults are not used but the actual scheduler config, but still.

7.
The scheduler config here is contradictory:
https://huggingface.co/docs/diffusers/main/api/pipelines/qwenimage#lora-for-faster-inference

base_shift and max_shift is set to the same value. That means there is no dynamic timestep shifting. But dynamic timestep shifting is enabled.

8.
https://github.com/huggingface/diffusers/blob/fc337d585309c4b032e8d0180bea683007219df1/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L141
Requires a AutoencoderKLQwenImage, not a AutoencoderKL

9.
The shape here is wrong:
https://github.com/huggingface/diffusers/blob/fc337d585309c4b032e8d0180bea683007219df1/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L391
At other points in the code, It's [B, C, F, H, W], not [B, F, C, H, W]. It doesn't have any effect currently with F always 1, because the shape is packed right after.

### Reproduction

if any of the points above need reproduction, let me know and I'll provide code

### Logs

```shell

```

### System Info

diffusers HEAD

### Who can help?

@yiyixuxu @DN6 @sayakpaul
CC @JingyaHuang @naykun because of this possibly related issue https://github.com/huggingface/diffusers/issues/12294

**Comment by dxqb** [comment]
sorry, CCed the wrong person
@JingyeChen-Canva

**Comment by dxqb** [comment]
You can find workarounds for some of these issues in this code, using the diffusers pipeline:
https://github.com/Nerogar/OneTrainer/blob/b09ef110d34232ff3abeade787457bea39d8da5f/modules/model/QwenModel.py#L161C1-L179C69
https://github.com/Nerogar/OneTrainer/blob/b09ef110d34232ff3abeade787457bea39d8da5f/modules/modelSetup/BaseQwenSetup.py#L133C1-L149C100

**Comment by sayakpaul** [comment]
I appreciate your thorough issue. For some of the points you mentioned (like 4) -- I welcome you to contribute fixes through PRs directly. We'd greatly appreciate that.

And I think we're confusing issues with redundancy, etc. Like for example, point 2 isn't a bug perse. So, IMO, it's just better to contribute an improvement PR directly and discuss things.

> base_shift and max_shift is set to the same value. That means there is no dynamic timestep shifting. But dynamic timestep shifting is enabled.

Mind you that it comes from:
https://github.com/ModelTC/Qwen-Image-Lightning/blob/342260e8f5468d2f24d084ce04f55e101007118b/generate_with_diffusers.py#L82C9-L97C10

We didn't come up with the config ourselves and the code example explicitly states that. Not sure, why it would be confusing.

For the other questions, I will let @naykun chime in as they would be the most knowledgeable to answer them.

**Comment by dxqb** [comment]
I can certainly help working on these issues, but for many of them I'd need some guidance first how it was intended.

Picking up on your examples, I noticed the documentation is missing in point 4, because the type hint doesn't match the type that is actually used, point 5. I don't know what was the intention so I cannot document point 4.

Same point 2: I would remove this code, *if* I am right that this doesn't do anything. Am I?

**Comment by jiqiujia** [comment]
any update here?

**Comment by sayakpaul** [comment]
Most of the issues here relate to some redundancy, which I believe the authors might have made intentionally to be a bit more future-proof. I wouldn't consider those as _issues_. And I would like to request you to be cognizant of that.

@dxqb 

> Picking up on your examples, I noticed the documentation is missing in point 4, because the type hint doesn't match the type that is actually used, point 5. I don't know what was the intention so I cannot document point 4.

You can fix the type annotation and take it from there. `img_shapes` are used for calculating the frequencies in RoPE.

I wouldn't remove any non-trivial code because I trust the authors have included them for a reason. 

Let's try to focus on actual logical bugs and not nitpicky code-related redundancies. I think this will make it easier for the future participants as well.

**Comment by dxqb** [comment]
> Most of the issues here relate to some redundancy, which I believe the authors might have made intentionally to be a bit more future-proof. I wouldn't consider those as _issues_. And I would like to request you to be cognizant of that.

> Let's try to focus on actual logical bugs and not nitpicky code-related redundancies. I think this will make it easier for the future participants as well.


There are severe issues with the Qwen Image code. Separate issues were created for those, and I compiled the remaining topics that don't deserve their own issue in this one.
So obviously, these will be less impactful - otherwise I would have created a separate issue for each.
Please let's not have a discussion about nitpicking when there was very good reason to question *everything* in this code.

https://github.com/huggingface/diffusers/issues/12294
https://github.com/huggingface/diffusers/issues/12344
https://github.com/huggingface/diffusers/issues/12075


> > Picking up on your examples, I noticed the documentation is missing in point 4, because the type hint doesn't match the type that is actually used, point 5. I don't know what was the intention so I cannot document point 4.
> 
> You can fix the type annotation and take it from there. `img_shapes` are used for calculating the frequencies in RoPE.

I do not know how to fix the type annotation, because I don't know what type was intended.
